api/v1beta2/configuration_types.go

```diff
@@ -105,12 +107,389 @@ type Property struct {
+	AccessKey                 *string `json:"access_key,omitempty" hcl:"access_key"`
+	UseAzureadAuth            *bool   `json:"use_azuread_auth,omitempty" hcl:"use_azuread_auth"`
+	ClientID                  *string `json:"client_id,omitempty" hcl:"client_id"`
+	ClientCertificatePassword *string `json:"client_certificate_password,omitempty" hcl:"client_certificate_password"`
+	// ClientCertificateSecret is a reference to a secret containing the client certificate
+	// It's used to replace the `client_certificate_path` in native hcl backend configuration as we cannot use local file paths in the kubernetes cluster.
+	ClientCertificateSecret *CurrentNSSecretSelector `json:"client_certificate_secret,omitempty" hcl:"ClientCertificateSecret,block"`
+	ClientSecret            *string                  `json:"client_secret,omitempty" hcl:"client_secret"`
+}
+
+// ConsulBackendConf defines all options supported by the Terraform `consul` backend type.
+// You can refer to https://www.terraform.io/language/settings/backends/consul for the usage of each option.
+type ConsulBackendConf struct {
+	Path        string  `json:"path" hcl:"path"`
+	AccessToken string  `json:"access_token" hcl:"access_token"`
+	Address     *string `json:"address,omitempty" hcl:"address"`
+	Schema      *string `json:"schema,omitempty" hcl:"schema"`
+	Datacenter  *string `json:"datacenter,omitempty" hcl:"datacenter"`
+	HTTPAuth    *string `json:"http_auth,omitempty" hcl:"http_auth"`
+	Gzip        *bool   `json:"gzip,omitempty" hcl:"gzip"`
+	Lock        *bool   `json:"lock,omitempty" hcl:"lock"`
+	// CAFileSecret is a reference to a secret containing the CA certificate
+	// It's used to replace the `ca_file` in native hcl backend configuration as we cannot use local file paths in the kubernetes cluster.
+	CAFileSecret *CurrentNSSecretSelector `json:"ca_file_secret,omitempty" hcl:"CAFileSecret,block"`
+	// CertFileSecret is a reference to a secret containing the client certificate
+	// It's used to replace the `cert_file` in native hcl backend configuration as we cannot use local file paths in the kubernetes cluster.
+	CertFileSecret *CurrentNSSecretSelector `json:"cert_file_secret,omitempty" hcl:"CertFileSecret,block"`
+	// KeyFileSecret is a reference to a secret containing the client key
+	// It's used to replace the `key_file` in native hcl backend configuration as we cannot use local file paths in the kubernetes cluster.
+	KeyFileSecret *CurrentNSSecretSelector `json:"key_file_secret,omitempty" hcl:"KeyFileSecret,block"`
+}
+
+// COSBackendConf defines all options supported by the Terraform `cos` backend type.
+// You can refer to https://www.terraform.io/language/settings/backends/cos for the usage of each option.
+type COSBackendConf struct {
+	SecretID  *string `json:"secret_id,omitempty" hcl:"secret_id"`
+	SecretKey *string `json:"secret_key,omitempty" hcl:"secret_key"`
+	Region    *string `json:"region,omitempty" hcl:"region"`
```

Comment:
Should we automatically set the region from a provider?

Reply:
I don't think it's necessary. Users may want to use a total "different" COS instance(may be completely unrelated to the environment involved in Configuration and Provider) to store their backend. I think we should keep it as it is and let the users to determine.

Reply:
You are right.